src/Workspaces/Core/Portable/Editing/DeclarationModifiers.cs

```diff
@@ -73,11 +73,12 @@ IMethodSymbol or
                 var field = symbol as IFieldSymbol;
                 var property = symbol as IPropertySymbol;
                 var method = symbol as IMethodSymbol;
+                var type = symbol as INamedTypeSymbol;
 
                 return new DeclarationModifiers(
                     isStatic: symbol.IsStatic,
                     isAbstract: symbol.IsAbstract,
-                    isReadOnly: field?.IsReadOnly == true || property?.IsReadOnly == true,
+                    isReadOnly: field?.IsReadOnly == true || property?.IsReadOnly == true || type?.IsReadOnly == true || method?.IsReadOnly == true,
```

Comment:
Could you please add a test to cover readonly method too?

Reply:
I did have one in here. For some reason it had me push this twice, and my test was in the last commit! Sorry for the confusion